dotnet/Services/ProductReviewService.cs

```diff
@@ -527,6 +542,34 @@ public async Task<ValidatedUser> ValidateUserToken(string token)
             return await this._productReviewRepository.ValidateUserToken(token);
         }
 
+        public async Task<dynamic> IsValidAuthUser()
+        {
+            if (string.IsNullOrEmpty(_context.Vtex.AdminUserAuthToken))
+            {
+                return HttpStatusCode.Unauthorized;
+            }
+
+            ValidatedUser validatedUser = null;
+            try {
+                validatedUser = await ValidateUserToken(_context.Vtex.AdminUserAuthToken);
+            }
+            catch (Exception ex)
+            {
+                _context.Vtex.Logger.Error("ModerateReview", null, "Error fetching user", ex);
+                return HttpStatusCode.BadRequest;
+            }
+
+            bool hasPermission = validatedUser != null && validatedUser.AuthStatus.Equals("Success");
+            
+            if (!hasPermission)
+            {
+                _context.Vtex.Logger.Error("ModerateReview", null, "User Does Not Have Permission");
```

Comment:
Let's make this a warning instead of an error since the problem is with the user, not with the app